### PR TITLE
Show info when game cannot be loaded

### LINKED ISSUE #339: Attempting to load game data that leads to an invalid state should jump to the last valid move in the history
Currently if you try to load invalid game data, the hotseat game will refuse to load/render, with a console error that the game constructor has failed. It would be nice if it would load at the correct action index to render the last valid game state, so that the error can be produced by clicking forward one step in the history, and the game state can be inspected.

### PR DESCRIPTION
**Pull request by jenf**
Fixes #339

(This'll be needed in the opal pinning for hotseat games)

The only thing I couldn't quite work out how to do is to be able to disable it when running in development, or a way of rethrowing the exception after the render.

### CODE REVIEW

File: lib/engine/game/base.rb [hunk 1]
```diff
@@ -305,6 +305,10 @@ def process_action(action)
         @actions << action
         next_round! while @round.finished? && !@finished
         self
+      rescue GameError => e
+        # Fill in the action id
+        e.action_id = current_action_id
+        raise
```
Comment (tobymao) [inline lib/engine/game/base.rb:311]: should we raise game error?

Comment (jenf) [reply lib/engine/game/base.rb:311]: I thought "raise" bubbled the current exception up

Comment (HQ063) [reply lib/engine/game/base.rb:311]: it does

Comment (jenf) [reply lib/engine/game/base.rb:311]: Toby, when you have a mo, could you rereview?

Comment (tobymao) [reply lib/engine/game/base.rb:311]: so this doesn't in dev?

Comment (tobymao) [reply lib/engine/game/base.rb:311]: do you have access to rack env?

Comment (jenf) [reply lib/engine/game/base.rb:311]: it's be in opal world, so no. but i could pass it in somehow.

Comment (tobymao) [reply lib/engine/game/base.rb:311]: can you just pass it in as a need

**Pull request merged**